src/jit/codegenarm64.cpp

```diff
@@ -9396,6 +9388,38 @@ void                CodeGen::genArm64EmitterUnitTests()
 }
 #endif // defined(DEBUG)
 
+
+//------------------------------------------------------------------------------------------------ //
+// getFirstArgWithStackSlot - returns the first argument with stack slot on the caller's frame.
+//
+// Return value:
+//    The number of the first argument with stack slot on the caller's frame.
+
+unsigned
+CodeGen::getFirstArgWithStackSlot()
+{
+    // Iterate over all the local variables in the lclvartable.
+    // They contain all the implicit argumets - thisPtr, retBuf,
+    // generic context, PInvoke cookie, var arg cookie,no-standard args, etc.
+    unsigned baseVarNum = 0;
+    LclVarDsc* varDsc = nullptr;
+    for (unsigned i = 0; i < compiler->lvaCount; i++)
+    {
+        varDsc = &(compiler->lvaTable[i]);
+
+        // We are iterating over the arguments only.
+        assert(varDsc->lvIsParam);
+
+        if (varDsc->lvArgReg == REG_STK)
```

Comment:
The reason why assert on line 9411 wasn't fired thought we used lvaCount as loop limit is that this routine was called while generating code for GT_PUTARG_STK and that by that time fast tail call logic has established that caller method also has a stack arg (and hence tail call possible).  Since caller method is guaranteed to have at least one arg passed on stack, within 0 to compiler->info.compArgsCount, the above loop will find such an arg.

On the other hand if the above routine was called in any other context could lead to an assert.